presto-tests/src/main/java/com/facebook/presto/tests/statistics/MetricComparator.java

```diff
@@ -65,21 +67,45 @@ private MetricComparator() {}
         return metricComparisons.build();
     }
 
-    private static List<Optional<?>> getEstimatedValues(List<Metric<?>> metrics, String query, DistributedQueryRunner runner)
+    private static List<Optional<?>> getEstimatedValues(List<Metric<?>> metrics, String query, QueryRunner runner)
     {
         return transaction(runner.getTransactionManager(), runner.getAccessControl())
                 .singleStatement()
                 .execute(runner.getDefaultSession(), session -> {
-                    String queryId = runner.executeWithQueryId(session, query).getQueryId();
-                    Plan queryPlan = runner.getQueryPlan(new QueryId(queryId));
-                    OutputNode outputNode = (OutputNode) queryPlan.getRoot();
-                    PlanNodeStatsEstimate outputNodeStats = calculateStats(outputNode, runner.getStatsCalculator(), session, queryPlan.getTypes());
-                    StatsContext statsContext = buildStatsContext(queryPlan, outputNode);
-                    List<Optional<?>> estimatedValues = getEstimatedValues(metrics, outputNodeStats, statsContext);
-                    return estimatedValues;
+                    if (runner instanceof DistributedQueryRunner) {
```

Comment:
Instead of this if, maybe `QueryRunner` should have `getPlan` method

Reply:
To facilitate this, i added some commits before this change: `Change type of ResultWithQueryId.queryId to QueryId`, `Add createPlan() method to QueryRunner interface`